File: Baekjoon/Problem_2000~2999/Problem_2293/Problem_2293.cpp
#include <cstdio>

const int MAXN = 100;
const int MAXK = 10000;

using namespace std;

int main() {
    int n, k; scanf("%d %d", &n, &k);

    int coins[MAXN];
    for (int i = 0; i < n; i++) scanf("%d", &coins[i]);

    int dp[MAXK+1] = { 0, };
    dp[0] = 1;

    for (int i = 0; i < n; i++) {
        int coin = coins[i];

        for (int j = coin; j <= k; j++) {
            dp[j] += dp[j-coin];
        }
    }

    
    printf("%d\n", dp[k]);

    return 0;
}

File: Baekjoon/Problem_2000~2999/Problem_2293/Problem_2293.py
import sys

input = lambda: sys.stdin.readline().rstrip()
    
if __name__ == "__main__":
    n, k = map(int, input().split())
    
    coins = sorted([ int(input()) for _ in range(n) ])
    
    dp = [ 0 ] * (k+1)
    dp[0] = 1
    
    for coin in coins:
        for i in range(coin, k+1):
            dp[i] += dp[i-coin]
    
    print(dp[k])